src/Microsoft.AspNet.OData/Extensions/HttpRequestMessageProperties.cs

```diff
@@ -104,9 +104,9 @@ public ODataPath Path
                 object totalCount;
                 if (_request.Properties.TryGetValue(TotalCountKey, out totalCount))
                 {
-                    // Fairly big problem if following cast fails. Indicates something else is writing properties with
-                    // names we've chosen. Do not silently return null because that will hide the problem.
-                    return (long)totalCount;
+                    // TotalCount is defined as null-able long, allowed to be set as null, so it should be
```

Comment:
It seems like we restricted setting total count as null intentionally. I think things can get messy if some uses the same property name which can also be set to null.